scala/tempo/src/main/scala/com/databrickslabs/tempo/rangeStats.scala

```diff
@@ -0,0 +1,61 @@
+package com.databrickslabs.tempo
+
+import org.apache.spark.sql.expressions.{Window, WindowSpec}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{Column, DataFrame}
+
+/**
+  * The following object has methods for adding rolling statistics based on the timestamp as the base point. The parameter to range stats is a number of seconds to look back to aggregate statistics such as mean, count, sum, min, max, etc.
+  */
+object rangeStats {
+
+  val SUMMARY_FUNCS: List[(String, String => Column)] = List(
+    ("mean_", {c: String => mean(c)}),
+    ("count_",{c: String => count(c)}),
+    ("sum_" , {c: String => sum(c)}),
+    ("min_" , {c: String => min(c)}),
+    ("max_" ,{c: String => max(c)}),
+    ("stddev_" ,{c: String => stddev(c)}))
+
+  // For functions that are derived from SUMMARY_FUNCS (and do not need a window function)
+  val DERIVE_FUNCS: List[(String, String => Column)] = List(
+    ("zscore_", {c: String => (col(c) - col("mean_" + c)) / col("stddev_" + c)}))
+
+  // helper function to get aggregation stats for all agg functions in SUMMARY_FUNCS for a given metric column
+  def doSummarize(df: DataFrame, colName: String, window_spec: WindowSpec): DataFrame  = {
+    SUMMARY_FUNCS.foldLeft(df)((df, f) => df.withColumn(f._1 + colName, f._2(colName).over(window_spec)))
+  }
+
+  def doDerive(df: DataFrame, colName: String): DataFrame = {
+    DERIVE_FUNCS.foldLeft(df)((df,f) => df.withColumn(f._1 + colName, f._2(colName)))
+  }
+
+  def rangeStatsExec(
```

Comment:
Some of these params, `sumarize_type` and `rangeBackWindowSecs` don't seem to be used, is this just placement for later ehancements?